File: src/backend/schemas/supplement.py
"""영양제(건강기능식품) 관련 API 스키마."""

from pydantic import BaseModel


class SupplementSearchItem(BaseModel):
    """영양제 검색 결과 아이템."""

    id: int
    product_name: str
    slug: str
    company: str | None = None
    main_ingredient: str | None = None
    category: str | None = None

    model_config = {"from_attributes": True}


class SupplementDetail(BaseModel):
    """영양제 상세 정보."""

    id: int
    product_name: str
    slug: str
    company: str | None = None
    registration_no: str | None = None
    main_ingredient: str | None = None
    ingredients: list[dict] | None = None
    functionality: str | None = None
    precautions: str | None = None
    intake_method: str | None = None
    category: str | None = None
    source: str | None = None

    model_config = {"from_attributes": True}

File: src/web/src/types/supplement.ts
/** 영양제(건강기능식품) 관련 타입 정의 — 백엔드 스키마 1:1 매칭. */

export interface SupplementSearchItem {
  id: number;
  product_name: string;
  slug: string;
  company: string | null;
  main_ingredient: string | null;
  category: string | null;
}

export interface SupplementDetail {
  id: number;
  product_name: string;
  slug: string;
  company: string | null;
  registration_no: string | null;
  main_ingredient: string | null;
  ingredients: Record<string, unknown>[] | null;
  functionality: string | null;
  precautions: string | null;
  intake_method: string | null;
  category: string | null;
  source: string | null;
}
